src/button/button-group.scss

```diff
@@ -5,72 +5,79 @@
 @mixin iui-button-group {
   display: inline-flex;
   align-items: center;
+  isolation: isolate;
 
-  > .iui-button {
-    &:not(.iui-borderless) {
-      border-radius: 0;
-    }
-
-    &:first-of-type:not(.iui-borderless) {
-      border-top-left-radius: $iui-border-radius;
-      border-bottom-left-radius: $iui-border-radius;
-    }
+  $childSelector: 'input, button'; // target nested inputs and buttons
 
-    &:last-of-type:not(.iui-borderless) {
-      border-top-right-radius: $iui-border-radius;
-      border-bottom-right-radius: $iui-border-radius;
-    }
+  > * {
+    #{$childSelector} {
+      &:hover,
+      &:focus,
+      &:focus-within {
+        position: relative;
+        z-index: 1;
+      }
 
-    &.iui-default:not(:last-of-type) {
-      border-right: none;
-    }
+      &:disabled {
+        position: relative;
+        z-index: -1;
+      }
 
-    &.iui-default {
-      // Border trick for grouped default & disabled buttons
-      &:not(:disabled) {
-        + :disabled {
+      // Adds stripe above active button
+      &.iui-active {
+        &::after {
+          content: '';
+          position: absolute;
+          height: $iui-xxs;
+          opacity: 1;
+          top: $iui-xxs;
+          left: $iui-xxs;
+          width: calc(100% - #{$iui-xs});
           @include themed {
-            border-left-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+            background-color: t(iui-color-foreground-primary);
           }
         }
 
-        &:hover + .iui-button,
-        &:active + .iui-button,
-        &:focus-visible + .iui-button {
-          @include themed {
-            border-left-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-1));
+        &:disabled {
+          &::after {
+            @include themed {
+              background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-5));
+            }
           }
         }
       }
 
-      // Border between 2 side-by-side disabled buttons
-      &:disabled + .iui-button:disabled {
-        @include themed {
-          border-left-color: t(iui-color-background-4);
-        }
+      // Reset border radius only for non-borderless items
+      &:not(.iui-borderless) {
```

Comment:
Why not omiting borderless?

Reply:
borderless buttons should keep using their normal radius.

It's the same as before
![image](https://user-images.githubusercontent.com/9084735/145420236-b7299cdc-7de7-4dfd-b10d-aff22ef89105.png)